clients/logger/logger.go

```diff
@@ -152,6 +130,35 @@ func newFileWriter(logTarget string) (io.Writer, error) {
 	return &fileWriter, nil
 }
 
+func (lc *edgeXLogger) initLogger(owningServiceName string, isRemote bool, logTarget, logLevel string) {
```

Comment:
Violates single responsibility principle.

Your factory functions are implementation. They can leverage a new public method (which is not part of the LoggingClient contract) on the receiver to return what you need.

Let me know if you want to pair code this.

Reply:
I disagree that it violates the SRP. This is a file that creates loggers. After my changes are merged, it will be a file that creates loggers. I'm interested to hear your take on why it isn't.

Further, I think your suggestion is adding a lot of complexity to a relatively straightforward solution to a small problem. What is the benefit of adding a new exported method? Who do we expect to consume that method? What is gained by doing this? With my proposed solution, we have one new exported function, one new unexported method, it is clear what the exported function does and its use case versus the existing exported function, and the existing exported function does not have any changed behavior. With the addition of the unexported method, there is no code duplication to worry about.

Unless I'm misunderstanding and you're suggesting we use a different name other than `NewClient` because that is part of the defined interface, then, sure. I don't have a problem with something like that.

Reply:
Resolved on call.